src/qibolab/platform.py

```diff
@@ -339,6 +339,14 @@ def _execute(self, method, sequences, options, **kwargs):
         if options.relaxation_time is None:
             options = replace(options, relaxation_time=self.relaxation_time)
 
+        duration = 0
+        if isinstance(sequences, list):
+            duration = sum(seq.duration for seq in sequences)
+        else:
+            duration = sequences.duration
```

Comment:
`duration = 0` is useless, because it gets overwritten in any case.

However, since you'd like to express the idea of defining a variable anyhow, in one way or another, `if` expressions are best suited for the purpose:
```suggestion
        duration = sum(seq.duration for seq in sequences) if isinstance(sequences, list) else sequences.duration
```

In this way, the variable definition is outside the `if` block, and only the value is conditional.